osquery/core/windows/process.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <vector>
+#include <sstream>
+#include <signal.h>
+#include <sys/types.h>
+
+#include "osquery/core/process.h"
+
+extern char **environ;
+
+namespace {
+osquery::PlatformPidType __declspec(nothrow) duplicateHandle(osquery::PlatformPidType src) {
+  osquery::PlatformPidType handle = osquery::kInvalidPid;
+
+  if (src != osquery::kInvalidPid) {
+    if (!::DuplicateHandle(GetCurrentProcess(),
+                           src,
+                           GetCurrentProcess(),
+                           &handle,
+                           0,
+                           FALSE,
+                           DUPLICATE_SAME_ACCESS)) {
+      handle = osquery::kInvalidPid;
+    }
+  }
+  return handle;
+}
+}
+
+namespace osquery {
+
+PlatformProcess::PlatformProcess(PlatformPidType id) { 
+  id_ = duplicateHandle(id);
+}
+
+PlatformProcess::PlatformProcess(const PlatformProcess& src) {
+  id_ = duplicateHandle(src.nativeHandle());
+}
+
+PlatformProcess::PlatformProcess(PlatformProcess&& src) {
+  id_ = kInvalidPid;
+  std::swap(id_, src.id_);
+}  
+
+PlatformProcess::~PlatformProcess() { 
+  if (id_ != kInvalidPid) {
+    ::CloseHandle(id_);
+    id_ = kInvalidPid;
+  }
+}
+
+PlatformProcess& PlatformProcess::operator=(const PlatformProcess& process) {
+  id_ = duplicateHandle(process.nativeHandle());
+  return *this;
+}
+
+bool PlatformProcess::operator==(const PlatformProcess& process) const {
+  return (::GetProcessId(nativeHandle()) == ::GetProcessId(process.nativeHandle()));
+}
+
+bool PlatformProcess::operator!=(const PlatformProcess& process) const {
+  return (::GetProcessId(nativeHandle()) != ::GetProcessId(process.nativeHandle()));
+}
+
+int PlatformProcess::pid() const {
+  return ::GetProcessId(id_);
+}
+
+bool PlatformProcess::kill() const {
+  if (id_ == kInvalidPid) {
+    return false;
+  }
+  
+  return ::TerminateProcess(id_, 0);
+}
+
+PlatformProcess PlatformProcess::launchWorker(const std::string& exec_path, const std::string& name) {
+  ::STARTUPINFOA si = { 0 };
+  ::PROCESS_INFORMATION pi = { 0 };
+  
+  si.cb = sizeof(si);
+  
+  // TODO(#1991): We currently do not sanitize or check for bad characters in for the worker name. Names
```

Comment:
Will the name have `\` and `"` in it? I think I'd rather just strip all non-alphanumeric and space instead.